core/templates/pages/learner-dashboard-page/learner-dashboard-icons.component.spec.ts

```diff
@@ -0,0 +1,640 @@
+        incomplete_exploration_ids: [],
+        incomplete_collection_ids: [],
+        completed_exploration_ids: [],
+        completed_collection_ids: [],
+        exploration_playlist_ids: [],
+        collection_playlist_ids: []
+      });
+
+    let result = component.belongsToCompletedActivities();
+
+    tick();
+    fixture.detectChanges();
+
+    expect(result).toBe(false);
+  }
+  ));
+
+  it('should return false if the curent collection belongs to the' +
+    ' completed playlist', fakeAsync(() => {
+    component.activityType = 'collection';
+    component.activityId = '1';
+    component.learnerDashboardActivityIds = LearnerDashboardActivityIds
+      .createFromBackendDict({
+        incomplete_exploration_ids: [],
+        incomplete_collection_ids: [],
+        completed_exploration_ids: [],
+        completed_collection_ids: [],
+        exploration_playlist_ids: [],
+        collection_playlist_ids: []
+      });
+
+    let result = component.belongsToCompletedActivities();
+
+    tick();
+    fixture.detectChanges();
+
+    expect(result).toBe(false);
+  }
+  ));
+
+  it('should return false if the curent exploration belongs to the' +
+    ' incomplete playlist', fakeAsync(() => {
+    component.activityType = 'exploration';
+    component.activityId = '1';
+    component.learnerDashboardActivityIds = LearnerDashboardActivityIds
+      .createFromBackendDict({
+        incomplete_exploration_ids: [],
+        incomplete_collection_ids: [],
+        completed_exploration_ids: [],
+        completed_collection_ids: [],
+        exploration_playlist_ids: [],
+        collection_playlist_ids: []
+      });
+
+    let result = component.belongsToIncompleteActivities();
+
+    tick();
+    fixture.detectChanges();
+
+    expect(result).toBe(false);
+  }
+  ));
+
+  it('should return false if the curent collection belongs to the' +
+    ' incomplete playlist', fakeAsync(() => {
+    component.activityType = 'collection';
+    component.activityId = '1';
+    component.learnerDashboardActivityIds = LearnerDashboardActivityIds
+      .createFromBackendDict({
+        incomplete_exploration_ids: [],
+        incomplete_collection_ids: [],
+        completed_exploration_ids: [],
+        completed_collection_ids: [],
+        exploration_playlist_ids: [],
+        collection_playlist_ids: []
+      });
+
+    let result = component.belongsToIncompleteActivities();
+
+    tick();
+    fixture.detectChanges();
+
+    expect(result).toBe(false);
+  }
+  ));
+
+  it('should open a modal for removing exploration from' +
+    ' learner playlist', fakeAsync(() => {
+    let activityId = '1';
+    let activityTitle = 'Title';
+    let activityType = 'exploration';
+
+    const learnerPlaylistSpy =
+      spyOn(learnerPlaylistBackendApiService, 'removeFromLearnerPlaylistModal')
+        .and.returnValue(null);
+
+    component.removeFromLearnerPlaylist(
+      activityId, activityTitle, activityType);
+
+    tick();
```

Comment:
In all these tests why is tick() needed?

Reply:
Removed!
tick() will make sure to flush if any microtasks, macrotasks exist before assertion.
But, It is redundant in the synchronous activities.